cg/meta/archive/archive.py

```diff
@@ -89,31 +89,45 @@ def archive_spring_files_and_add_archives_to_housekeeper(
             else:
                 LOG.info(f"No files to archive for location {archive_location}.")
 
-    def retrieve_samples(self, sample_internal_ids: list[str]) -> None:
-        """Retrieves the archived spring files for a list of samples."""
-        samples: list[Sample] = [
-            self.status_db.get_sample_by_internal_id(sample_internal_id)
-            for sample_internal_id in sample_internal_ids
-        ]
-        samples_and_destinations: list[SampleAndDestination] = self.join_destinations_and_samples(
-            samples
+    def retrieve_case(self, case_id: str) -> None:
+        """Submits jobs to retrieve any archived files belonging to the given case, and updates the Archive entries
+        with the retrieval job id."""
+        case: Case = self.status_db.get_case_by_internal_id(case_id)
+        files_to_retrieve: list[File] = self.get_files_to_retrieve(case)
+        self.retrieve_files_from_archive_location(
+            files_and_samples=self.add_samples_to_files(files=files_to_retrieve),
+            archive_location=case.customer.data_archive_location,
         )
-        for archive_location in ArchiveLocations:
-            filtered_samples: list[SampleAndDestination] = filter_samples_on_archive_location(
-                samples_and_destinations=samples_and_destinations,
-                archive_location=archive_location,
+
+    def get_files_to_retrieve(self, case: Case):
+        return [
+            file
+            for file in self.get_archived_spring_files_for_case(case)
+            if file.archive.retrieval_task_id is None
```

Comment:
```suggestion
            if not file.archive.retrieval_task_id
```

Reply:
I mean, we could have the very unfortunate instance where `file.archive.retrieval_task_id` is 0, whereupon this change would break the functionality. So I prefer having it as is even though it is slightly uglier.